src/component/helpers/array-utils.ts

```diff
@@ -55,10 +55,13 @@ export function ensureIsArray<T>(elements: (T | string)[] | T | string, acceptEm
 export function filter<T extends string>(arrayToFilter: T[], suffix: string, options?: FilterParameter): T[] {
   let pattern = '';
   if (options?.startingWith) {
+    // eslint-disable-next-line unicorn/prefer-spread -- pattern is a string not an array
     pattern = pattern.concat(`^(${options.startingWith}).*`);
   } else if (options?.notStartingWith) {
+    // eslint-disable-next-line unicorn/prefer-spread -- pattern is a string not an array
     pattern = pattern.concat(`^(?!(${options.notStartingWith})).*`);
   }
+  // eslint-disable-next-line unicorn/prefer-spread -- pattern is a string not an array
   pattern = pattern.concat(`${suffix}$`);
```

Comment:
**suggestion:** inspired by a comment in the unicorn issues.
Tested locally:
- ✔️ lint
- ✔️ tests (unit + integration)

```suggestion
  const patterns: string[] = [];
  if (options?.startingWith) {
    patterns.push(`^(${options.startingWith}).*`);
  } else if (options?.notStartingWith) {
    patterns.push(`^(?!(${options.notStartingWith})).*`);
  }
  patterns.push(`${suffix}$`);
  const pattern = patterns.join('');
```